File: comms/bulk-messaging/sender-pools/list-sender-pool-operations/list-sender-pool-operations.rb
require "twiliocomms"

client = TwilioComms::Client.new(
  account_id: "TWILIO_ACCOUNT_SID",
  auth_token: "TWILIO_AUTH_TOKEN"
)

client.sender_pools.list_operations(
  start_date: "2024-01-15T09:30:00Z",
  end_date: "2024-01-15T09:30:00Z",
  status: "PROCESSING",
  page_token: "pageToken",
  page_size: 50
)

File: comms/bulk-messaging/sender-pools/list-sender-pool-operations/list-sender-pool-operations.php
<?php

namespace Example;

use Twilio\Comms\TwilioComms;
use Twilio\Comms\SenderPools\Requests\SenderPoolsListOperationsRequest;
use DateTime;
use Twilio\Comms\Types\OperationStatus;

$client = new TwilioComms(
    accountId: 'TWILIO_ACCOUNT_SID',
    authToken: 'TWILIO_AUTH_TOKEN',
);
$client->senderPools->listOperations(
    new SenderPoolsListOperationsRequest([
        'startDate' => new DateTime('2024-01-15T09:30:00Z'),
        'endDate' => new DateTime('2024-01-15T09:30:00Z'),
        'status' => OperationStatus::Processing->value,
        'pageToken' => 'pageToken',
        'pageSize' => 50,
    ]),
);
